src/core/tree_methods.test.ts

```diff
@@ -0,0 +1,77 @@
+import * as tree from './tree_methods'
+import * as _ from 'lodash'
+
+let root;
+
+let treeRoot = new tree.TreeNode("korzen", false);
+let note1 = new tree.TreeNode("notka1", false);
+let note2 = new tree.TreeNode("notka2", false);
+let note3 = new tree.TreeNode("notka3", false);
+let note21 = new tree.TreeNode("notka1od2", false);
+let note22 = new tree.TreeNode("notka2od2", false);
+let note221 = new tree.TreeNode("notka1od22", false);
+let note222 = new tree.TreeNode("notka2od22", false);
+let note31 = new tree.TreeNode("notka1od3", false);
+
+treeRoot.add(note1);
+treeRoot.add(note2);
+treeRoot.add(note3);
+note2.add(note21);
+note2.add(note22);
+note22.add(note221);
+note22.add(note222);
+note3.add(note31);
+
+beforeEach(() => {
```

Comment:
Can we duplicate this line in each test?
Using `beforeEach` is a smell and IMHO unneeded here.

Reply:
Setting up environment before each test is what beforeEach does. Not sure what you mean.